rendering/markdown.go

```diff
@@ -41,6 +41,21 @@ type highlightHTMLRenderer struct {
 	blackfriday.Renderer
 }
 
+// ListItem overrides the default ListItem render and adds support for GH-style 
+// checkboxes on service side. For the contents of the list item beyond the
+// checkbox prefix, the default Html.ListItem is called.
+func (h highlightHTMLRenderer) ListItem(out *bytes.Buffer, text []byte, flags int) {
+	switch {
+		case bytes.HasPrefix(text, []byte("[] ")):
+			text = append([]byte(`<input type="checkbox"></input>`), text[3:]...)
```

Comment:
This will not work without either replacing large parts of the markdown processor *or* messing around with internal data structures of it. The contents of the item (the text after the "[]") is processed by the upstream library and written to a buffer. *In theory* we could run the processing, then take the contents of the buffer (which is incremental and already contains data when given to ListItem()), modify it, empty out the original buffer and replace the data with the new postprocessed data.

So the problem here is, we can only do precondition modifications of the stream, not postcondition, so adding something *after* the content of the item is not possible without a lot of work. 

And it is also not possible to "trick" our way out here as the upstream ListItem() already adds the closing </li> to the buffer.